src/registrar/views/portfolios.py

```diff
@@ -492,138 +495,142 @@ def post(self, request, *args, **kwargs):
+
+        # Check if the email requestor has a valid email address
+        if not requestor.is_staff and requestor.email is not None and requestor.email.strip() != "":
+            requestor_email = requestor.email
+        elif not requestor.is_staff:
+            messages.error(self.request, "Can't send invitation email. No email is associated with your account.")
+            logger.error(
+                f"Can't send email to '{email}' on domain '{self.object}'."
+                f"No email exists for the requestor '{requestor.username}'.",
+                exc_info=True,
+            )
+            return None
+
+        # Check to see if an invite has already been sent
+        try:
+            invite = PortfolioInvitation.objects.get(email=email, portfolio=self.object)
+            # check if the invite has already been accepted
+            if invite.status == PortfolioInvitation.PortfolioInvitationStatus.RETRIEVED:
+                add_success = False
+                messages.warning(
+                    self.request,
+                    f"{email} is already a manager for this portfolio.",
+                )
+            else:
+                add_success = False
+                # else if it has been sent but not accepted
+                messages.warning(self.request, f"{email} has already been invited to this portfolio")
+        except Exception:
+            logger.error("An error occured")
+
+        try:
+            logger.debug("requestor email: " + requestor_email)
+
+            # send_templated_email(
+            #     "emails/portfolio_invitation.txt",
+            #     "emails/portfolio_invitation_subject.txt",
+            #     to_address=email,
+            #     context={
+            #         "portfolio": self.object,
+            #         "requestor_email": requestor_email,
+            #     },
+            # )
+        except EmailSendingError as exc:
+            logger.warn(
+                "Could not sent email invitation to %s for domain %s",
+                email,
+                self.object,
+                exc_info=True,
+            )
+            raise EmailSendingError("Could not send email invitation.") from exc
+        else:
+            if add_success:
+                messages.success(self.request, f"{email} has been invited.")
+
+    def _make_invitation(self, email_address: str, requestor: User):
+        """Make a Member invitation for this email and redirect with a message."""
+        try:
+            self._send_portfolio_invitation_email(email=email_address, requestor=requestor)
+        except EmailSendingError:
+            messages.warning(self.request, "Could not send email invitation.")
+        else:
+            # (NOTE: only create a MemberInvitation if the e-mail sends correctly)
+            PortfolioInvitation.objects.get_or_create(email=email_address, portfolio=self.object)
+        return redirect(self.get_success_url())
```

Comment:
Right now we have some code duplication with _send_portfolio_invitation_email, _send_domain_invitation_email and send_templated_email.  It would be great to somehow consolidate these, but for now I'm just trying to promote higher cohesion by separating the e-mail sending functionality from making invitation objects in the database.  This is debatable, but it is currently consistent with what we have elsewhere.  I figure any refactoring of the e-mail function should happen during the e-mail sending ticket.